### PR TITLE
feat: add job manager schema

### PR DESCRIPTION
**Pull request by netanelC**

### CODE REVIEW

File: schemas/infra/jobManager/forClients/v1.schema.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          "type": "number",
+          "description": "Polling interval time to queue task (in milliseconds)",
+          "default": 5000,
+          "x-env-value": "TASK_POLLING_INTERVAL_TIME"
+        },
+        "maxAttempts": {
+          "type": "number",
+          "description": "Max attempts of working on task",
+          "default": 3,
+          "x-env-value": "TASK_MAX_ATTEMPTS"
```
Comment (CptSchnitz) [inline schemas/infra/jobManager/forClients/v1.schema.json:44]: add validation

File: schemas/infra/jobManager/forClients/v1.schema.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      "type": "object",
+      "description": "Heartbeat manager configurations",
+      "required": ["url"],
+      "properties": {
+        "url": {
+          "type": "string",
+          "description": "The url of heartbeat-manager",
+          "x-env-value": "HEARTBEAT_URL"
+        },
+        "pingingIntervalTime": {
```
Comment (CptSchnitz) [inline schemas/infra/jobManager/forClients/v1.schema.json:58]: name should include milliseconds

**Review by CptSchnitz — CHANGES_REQUESTED** [review]
1. We need approval from all job-managers users
2. consider splitting job creator and worker

File: schemas/infra/jobManager/forClients/v1.schema.json (hunk 2 above)
Comment (netanelC) [reply schemas/infra/jobManager/forClients/v1.schema.json]: Added

File: schemas/infra/jobManager/job/v1.schema.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  "type": "object",
+  "required": ["types"],
+  "properties": {
+    "types": {
+      "type": "array",
+      "description": "The job types",
+      "items": {
+        "type": "string"
+      },
+      "x-env-value": "JOB_TYPES"
```
Comment (CptSchnitz) [inline schemas/infra/jobManager/job/v1.schema.json:13]: you can't use ENV for array for now

File: schemas/infra/jobManager/job/v1.schema.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  "description": "The job schema for job manager",
+  "type": "object",
+  "required": ["types"],
+  "properties": {
+    "types": {
+      "type": "array",
+      "description": "The job types",
+      "items": {
+        "type": "string"
+      },
```
Comment (CptSchnitz) [inline schemas/infra/jobManager/job/v1.schema.json:12]: add validation so there are no duplicates

File: schemas/infra/jobManager/task/v1.schema.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  "type": "object",
+  "required": ["types"],
+  "properties": {
+    "types": {
+      "type": "array",
+      "description": "The task types",
+      "items": {
+        "type": "string"
+      },
+      "x-env-value": "TASK_TYPES"
```
Comment (CptSchnitz) [inline schemas/infra/jobManager/task/v1.schema.json:13]: same comment as for job

File: schemas/infra/jobManager/sdk/v1.schema.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  "allOf": [
+    {
+      "type": "object",
+      "properties": {
+        "url": {
+          "type": "string",
+          "description": "The url of job-manager",
+          "format": "uri",
+          "x-env-value": "JOB_MANAGER_URL"
+        },
```
Comment (CptSchnitz) [inline schemas/infra/jobManager/sdk/v1.schema.json:15]: needs its own schema for reuse

**Review by CptSchnitz — CHANGES_REQUESTED** [review]
The difference between SDK and worker is not explained at all. We need better names as worker is also type of sdk

File: schemas/infra/jobManager/forClients/v1.schema.json (hunk 1 above)
Comment (netanelC) [reply schemas/infra/jobManager/forClients/v1.schema.json]: Added

File: schemas/infra/jobManager/sdk/v1.schema.json (hunk 6 above)
Comment (netanelC) [reply schemas/infra/jobManager/sdk/v1.schema.json:15]: Added base schema

File: schemas/infra/jobManager/job/v1.schema.json (hunk 3 above)
Comment (netanelC) [reply schemas/infra/jobManager/job/v1.schema.json]: Fixed

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**